ts/src/pro/binance.ts

```diff
@@ -293,8 +293,10 @@ export default class binance extends binanceRest {
         //         ]
         //     }
         //
-        const index = client.url.indexOf ('/stream');
-        const marketType = (index >= 0) ? 'spot' : 'contract';
+        // Note: not calling self.market(message['s'])['type'] because it's very slow (10x slower)
+        const isTestnetSpot = client.url.indexOf ('testnet') > 0;
+        const isSpot = client.url.indexOf ('/stream.binance.') > 0;
+        const marketType = (isTestnetSpot || isSpot) ? 'spot' : 'contract';
```

Comment:
not sure we can do this, what if you;re fetching a swap orderbook from the futures testnet?

Reply:
Hence the '.' in the end of the string.